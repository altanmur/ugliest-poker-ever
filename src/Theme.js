import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

export const theme = {
  colors: {
    indigo: '#4A0968',
    blackcurrant: '#27172F',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Theme;
