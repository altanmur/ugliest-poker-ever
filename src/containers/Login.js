import React from 'react';
import { Button, Input, MyLink, Panel } from '../components';

const Login = () => (
  <Panel>
    <Input placeholder="Nombre de usuario" name="uid" />
    <Input placeholder="ContraseÃ±a" name="passwd" />
    <Button>Ingresar</Button>
    <MyLink to="/register">No estas registrado?</MyLink>
  </Panel>
);

export default Login;
