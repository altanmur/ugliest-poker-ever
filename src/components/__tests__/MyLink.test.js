import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import MyLink from '../MyLink';
import Login from '../../containers/Login';
import { theme } from '../../Theme';

describe('MyLink Component', () => {
  it('should render correctly', () => {
    const component = shallow(<MyLink to={Login} theme={theme} />);

    expect(component).toMatchSnapshot();
  });
});
