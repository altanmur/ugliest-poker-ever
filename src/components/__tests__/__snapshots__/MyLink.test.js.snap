// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLink Component should render correctly 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #1b2f04;
  cursor: pointer;
  font-weight: bold;
  margin: 10px 0;
}

<Link
  className="c0"
  theme={
    Object {
      "colors": Object {
        "bahia": "#a8d90d",
        "bahiaGradient": "linear-gradient(0deg, #308a00 1%, #c3e80e 80%)",
        "blackcurrant": "#27172F",
        "christi": "#4fb30c",
        "indigo": "#4A0968",
        "limeade": "#3ea300",
        "milanoRed": "#bd2d02",
        "pineTree": "#1b2f04",
      },
    }
  }
  to={[Function]}
/>
`;
